crates/stc_ts_file_analyzer/src/analyzer/expr/call_new.rs

```diff
@@ -2211,9 +2211,25 @@ impl Analyzer<'_, '_> {
 
         let span = span.with_ctxt(SyntaxContext::empty());
 
-        let mut min_param: usize = params.iter().map(|v| &v.pat).map(count_required_pat).sum();
+        let min_param: usize = params
+            .iter()
+            .enumerate()
+            .filter_map(|(i, v)| {
+                if v.ty.contains_void() {
+                    let next_params = params[i + 1..params.len()].to_vec();
```

Comment:
Is `to_vec()` necessary? It allocates, so I'm curious if `&params[i + 1..params.len()]` will work